spec/valid_email2_spec.rb

```diff
@@ -59,6 +59,22 @@ class TestUserDisallowBlacklisted < TestModel
   validates :email, 'valid_email_2/email': { blacklist: true }
 end
 
+class TestUserDisallowBlacklistedWithDynamicArray < TestModel
+  validates :email, 'valid_email_2/email': { blacklist: ["test-dynamic-array.com"] }
+end
+
+class TestUserDisallowBlacklistedWithDynamicSet < TestModel
+  validates :email, 'valid_email_2/email': { blacklist: Set.new(["test-dynamic-set.com"]) }
+end
+
+class TestUserDisallowBlacklistedWithDynamicProc < TestModel
+  validates :email, 'valid_email_2/email': { blacklist: proc { ["test-dynamic-proc.com"] } }
+end
+
+class TestUserDisallowBlacklistedWithDynamicHash < TestModel
+  validates :email, 'valid_email_2/email': { blacklist: { active_record_scope: TestDynamicDomainModel, attribute: :domain } }
```

Comment:
The idea is for the active record scope you can do something like:

```ruby
blacklist: { active_record_scope: DomainList.where(blacklist: true), attribute: :domain }
```

And we'll do `DomainList.where(blacklist: true).exists?(domain: "foo.com")` as a performant query to check against.